core/src/services/onedrive/backend.rs

```diff
@@ -209,4 +315,75 @@ impl OnedriveBackend {
 
         self.client.send(req).await
     }
+
+    pub(crate) async fn onedrive_custom_put(
+        &self,
+        url: &str,
+        size: Option<usize>,
+        content_type: Option<&str>,
+        additional_headers: Option<HashMap<String, String>>,
```

Comment:
Please build header in `onedrive_put` by accepting `OpWrite`,

Reply:
This content length is different from that in `OpWrite`.

`onedrive_custom_put`'s `Content-Length` is for small chunks.

What do you suggest in this case?

Reply:
Please refer to gcs services. We should have two different upload functions and content length should set based on input bytes length.

Reply:
`GcsCore::gcs_upload_in_resumable_upload()`?

Reply:
Not sure whether it looks good to you?

```rs
async fn onedrive_chunked_upload(
        &self,
        url: &str,
        content_type: Option<&str>,
        offset: usize,
        chunk_end: usize,
        total_len: usize,
        body: AsyncBody,
    ) -> Result<Response<IncomingAsyncBody>>
```

Reply:
LGTM!